code/modules/projectiles/guns/ballistic/shotgun.dm

```diff
@@ -24,14 +24,46 @@
 	bolt_wording = "pump"
 	cartridge_wording = "shell"
 	tac_reloads = FALSE
-	weapon_weight = WEAPON_HEAVY
-
 	pb_knockback = 2
+	/// Keeps track of wielded status on the shotgun
+	var/wielded = FALSE
+
+/obj/item/gun/ballistic/shotgun/examine(mob/user)
+	. = ..()
+	. += "<span class='notice'>Use Ctrl-click to wield the gun.</span>"
+
+/obj/item/gun/ballistic/shotgun/Initialize()
+	. = ..()
+	RegisterSignal(src, COMSIG_TWOHANDED_WIELD, .proc/on_wield)
+	RegisterSignal(src, COMSIG_TWOHANDED_UNWIELD, .proc/on_unwield)
+
+/obj/item/gun/ballistic/shotgun/ComponentInitialize()
+	. = ..()
+	AddComponent(/datum/component/two_handed)
+
+///Triggered on wield of two handed item
+/obj/item/gun/ballistic/shotgun/proc/on_wield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+
+	wielded = TRUE
+
+///Triggered on unwield of two handed item
+/obj/item/gun/ballistic/shotgun/proc/on_unwield(obj/item/source, mob/user)
+	SIGNAL_HANDLER
+
+	wielded = FALSE
+
+/obj/item/gun/ballistic/shotgun/process_fire(atom/target, mob/living/user, message, params, zone_override, bonus_spread, forced)
```

Comment:
instead of doing this you should just make it fly out of the persons hand but still shoot.


Create a signal for shooting instead and make the component listen for it btw.

Reply:
It'd be better if you could make it drop things in both hands, the reason being it'd prevent you from just using two improvised shotguns and firing them both at once. They both fly out of your hands, but it doesn't really matter at that point; if you're adjacent to your target that's 12 buckshot pellets in them. If they aren't in softcrit they're a pimp bitchslap away from it.
I guess it could to chat something like "the intense recoil causes the [src] to whack you in the face!" and it gives your head a little complimentary brute damage?